lsst_dashboard/dataset.py

```diff
@@ -118,3 +132,125 @@ def fetch_visits_by_metric(self):
 
                 column_map = {'tractId': 'tract', 'visitId': 'visit', 'patchId': 'patch'}
                 self.visits_by_metric[filt][metric] = dd.read_parquet(filenames).rename(columns=column_map)
+
+
+
+class KartothekDataset():
+    """
+    USAGE:
+        d = KartothekDataset(path)
+        d.connect()
+        d.init_data()
+    """
+    def __init__(self, path, tracts=None, filters=None):
+        self.conn = None
+        self.path = Path(path)
+        self.coadd = {}
+        self.visits = None
+        self.visits_by_metric = {}
+        self.metadata = {}
+        self.metrics = []
+        self.failures = {}
+        self.flags = []
+        self.tracts = tracts
+        self.filters = filters if filters is not None else []
+
+    def connect(self):
+
+        self.parse_metadata_from_file()
+
+        print('-- read coadd table --')
+        self.fetch_coadd_table()  # currently ignoring forced/unforced
+
+        print('-- generate other metadata fields --')
+        self.post_process_metadata()
+
+        print('-- read visit data --')
+        self.fetch_visits_by_metric()
+
+        print('-- done with reads --')
+
+
+    def parse_metadata_from_file(self):
+        if self.path.joinpath(METADATA_FILENAME).exists():
+            self.metadata_path = self.path.joinpath(METADATA_FILENAME)
+        else:
+            self.metadata_path = Path(os.environ.get('LSST_META', os.curdir)).joinpath(self.path.name, METADATA_FILENAME)
+
+        with self.metadata_path.open('r') as f:
+            self.metadata = yaml.load(f, Loader=yaml.SafeLoader)
+            self.failures = self.metadata.get('failures', {})
+            if self.tracts is None:
+                self.tracts = list(set(x for v in self.metadata['visits'].values() for x in v.keys()))
+
+    def fetch_coadd_table(self, coadd_version='unforced'):
+        table = 'qaDashboardCoaddTable'
+        store = partial(get_store_from_url, 'hfs://' + str(self.path))
+        print(str(self.path))
+        predicates = [[('tract', 'in', self.tracts)]]
+        dataset = "coadd_{}".format(coadd_version)
+
+        coadd_df = read_dataset_as_ddf(predicates=predicates,
+                                       dataset_uuid=dataset,
+                                       store=store,
+                                       table='table').compute()
+
+        # hack label in ...
+        coadd_df['label'] = 'star'
+
+        self.coadd[table] = coadd_df
```

Comment:
Let's change the key here (and wherever else necessary) to be `dataset` rather than `table`, because the above `table` definition will not exist in the new format.